internal/rootcoord/proxy_client_manager.go

```diff
@@ -172,3 +176,102 @@ func (p *proxyClientManager) ReleaseDQLMessageStream(ctx context.Context, in *pr
 		ErrorCode: commonpb.ErrorCode_Success,
 	}, nil
 }
+
+func (p *proxyClientManager) InvalidateCredentialCache(ctx context.Context, request *proxypb.InvalidateCredCacheRequest) {
+	p.credLock.Lock()
+	defer p.credLock.Unlock()
```

Comment:
`credLock` and `lock` both protect `proxyClient`, which leads to data race. This lock shall be unified.

Reply:
done